src/IECoreArnold/Renderer.cpp

```diff
@@ -860,8 +869,10 @@ class ShaderCache : public IECore::RefCounted
 			return writeAccessor->second;
 		}
 
-		// Must not be called concurrently with anything.
-		void clearUnused()
+		// _Must_ be called before `render()` launches Arnold, and must not be called concurrently with anything.
+		// This removed shaders that are no longer in use, and ensures all `ShaderNetworkAlgo::NodeParameters`
```

Comment:
Quite picky, but the verb tense made me think twice - "This removed" should be "This removes"?

Reply:
Thanks - fixed in https://github.com/GafferHQ/gaffer/pull/5540/commits/823571699a8dddd1ab1312edc7f2cf2d1b5609d4.